lms/djangoapps/completion/handlers.py

```diff
@@ -21,9 +22,15 @@ def scorable_block_completion(sender, **kwargs):  # pylint: disable=unused-argum
     """
     if not waffle.waffle().is_enabled(waffle.ENABLE_COMPLETION_TRACKING):
         return
-    user = User.objects.get(id=kwargs['user_id'])
     course_key = CourseKey.from_string(kwargs['course_id'])
     block_key = UsageKey.from_string(kwargs['usage_id'])
+    store = modulestore()
```

Comment:
I'd rather find a way to do this without accessing the modulestore.  If there's a way to register metadata about a block just based on it's block_type, that would be ideal.

Reply:
@jcdyer I feel like there might be some sort of registry like that, but I can't find it. I may just be thinking of [`XBlockConfiguration`](https://github.com/edx/edx-platform/blob/e8a36957b1f732974260e7b9b42b9c25148b492c/common/djangoapps/xblock_django/models.py#L10).

If you have access to any XBlock runtime, you can use [`runtime.load_block_type`](https://github.com/edx/XBlock/blob/b0a6120208220c7ffffac5db189db03f7f26ee12/xblock/runtime.py#L600), but since that's probably not the case, you could probably get by with [`XBlock.load_class`](https://github.com/edx/XBlock/blob/0496f693352d962f5b7c2482b3fc410c599e3505/xblock/plugin.py#L80-L100) ([example](https://github.com/edx/edx-platform/blob/e8a36957b1f732974260e7b9b42b9c25148b492c/openedx/core/djangoapps/common_views/xblock.py#L21-L24)).